src/query/dblink_scan.c

```diff
@@ -249,17 +355,28 @@ dblink_open_scan (DBLINK_SCAN_INFO * scan_info, char *conn_url, char *user_name,
 int
 dblink_close_scan (DBLINK_SCAN_INFO * scan_info)
 {
-  T_CCI_ERROR err_buf;
   int error;
+  T_CCI_ERROR err_buf;
 
-  if ((error = cci_close_req_handle (scan_info->stmt_handle)) < 0)
+  /*  note: return NO_ERROR even though the connection or stmt handle is not valid */
+
+  if (scan_info->stmt_handle >= 0)
     {
-      return error;
+      if ((error = cci_close_req_handle (scan_info->stmt_handle)) < 0)
+	{
+	  cci_get_err_msg (error, err_buf.err_msg, sizeof (err_buf.err_msg));
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_DBLINK, 1, err_buf.err_msg);
+	  return S_ERROR;
```

Comment:
Even if the close of the stmt handle fails,
Closing conn_handle would be better if possible.

Reply:
Ok, thanks